scripts/zones/Uleguerand_Range/mobs/Mountain_Worm_NM.lua

```diff
@@ -0,0 +1,29 @@
+-----------------------------------
+-- Area: Uleguerand Range
+--  Mob: Mountain Worm NM
+--  https://www.bg-wiki.com/ffxi/Mountain_Worm_(NM)
+-- TODO allow deaggro based on distance (core CMobEntity::CanDeaggro() forces NM and Battlefield mobs to never stop chasing)
+-- TODO confirm if despawns on lost aggro?
+-----------------------------------
+local entity = {}
+
+entity.onMobSpawn = function(mob)
+    mob:setMagicCastingEnabled(false)
+    mob:setMod(xi.mod.REGEN, mob:getMaxHP() / 100)
+end
+
+entity.onMobFight = function(mob, target)
+    if mob:checkDistance(target) > mob:getModelSize() + 3 then
```

Comment:
kinda surprised the attempt at checking size wasn't baked into distance checks already (it'd be wrong, but its something i expect the mistake to be made in). 

`getModelSize()` really needs either renamed or how we determine size needs reworked for mobs, because the value stored in `m_ModelRadius` is NOT the physical size of the model on screen: its the "melee range" of the monster stored in `mob_family_system` that currently is not related to its actual size. you can set crazy values there and have a tiny little chigoe that is able to melee hit you and be hit in return from 30 yalms away. And that's the value you are adding 3 to and including here.

The mobs actual size is set in a few bits of a mask that have different purposes depending on which type of entity model it is. In the case of this worm, there are 3 single bits that influence how large it appears in the client. These are currently in the 1st byte of the entityFlags value stored in `mob_pools`

tl;dr You should probably add a comment here about how because he shares a family with the normal mountain worm but is a wee bit larger you had to expand this a bit. At least if I'm correct about that being the case anyway. If I'm not, prolly should drop the +3 then.

Reply:
Btw I didn't add this because of his size, I did it because that's the exact logic in core to determine if a mob will move or auto attack

Reply:
so you remade logic related to movement for a mob that _doesn't move._ 🤔 

I know that spot in the core, I've touched it myself. its...less than optimal and fudges on the side of making sure players can actually hit things they are standing in front of despite how fat king behemoths rear end may be. Now I'm thinking again about the mobmod line of thought I was having, maybe that bridge _should_ be crossed sooner rather than later, instead of running a distance check a and mob radius check then attempting to set its casting ability every 240ms.


I think you really wanted "can I hit the player?". I really hope the logic in the **draw in** code that works on that very premise doesn't do the same thing.

_looks_

```cpp
if (currentDistance >= PMob->GetMeleeRange() * 2 && battleutils::DrawIn(PTarget, PMob, PMob->GetMeleeRange() - 0.2f))
```

```cpp
float CBattleEntity::GetMeleeRange() const
{
    return m_ModelRadius + 3.0f;
}
```

_**"Son of a...**_

😭 we really can't win can we. We may as well lua bind `getMeleeRange()` and save some future steps at least, because _some day_ those size bits are going to be correctly used and then melee range will get updated to account for them.

Reply:
If we want that binding I can add it, sure

Reply:
> If we want that binding I can add it, sure

yeah lets do that for now theres bound to be other times we want to know melee range in lua

Reply:
Or perhaps I could remove this logic completely and let [this PR](https://github.com/LandSandBoat/server/pull/5313/files) handle it

Reply:
I'd forgotten about that, last I recall it needed to sort out which worms did and didn't do this behavior. But if you wnat to leave the casting behavior to that, that's find by me as well.

Reply:
I'll remove the mob fight logic altogether and watch for that PR to include this NM. Ezpz